pygame_gui/elements/ui_drop_down_menu.py

```diff
@@ -257,15 +260,21 @@ def process_event(self, event: pygame.event.Event) -> bool:
             self.drop_down_menu_ui.selected_option = selection
             self.should_transition = True
 
+            # If they provided a dictionary with a callable, use that.
+            selected_text = self.drop_down_menu_ui.selected_option
+            if selected_text and self.options_dict and self.options_dict[selected_text]:
+                self.options_dict[selected_text]()
+                return True     # In this case, consume the event.
```

Comment:
I think we should be consistent in consuming the `UI_SELECTION_LIST_NEW_SELECTION` event whether we post the `UI_DROP_DOWN_MENU_CHANGED` event or not. 

I actually think your approach is correct and we **should** be consuming the `UI_SELECTION_LIST_NEW_SELECTION` event here. There is no real loss in available functionality as users will still have their call backs or the `UI_DROP_DOWN_MENU_CHANGED` events if they want to ping a sound effect or something here. This is really an internal event for the drop down and not consuming it could potentially lead to confusion or double event triggering.